spec/datadog/profiling/stack_recorder_spec.rb

```diff
@@ -1027,6 +1027,17 @@ def sample_allocation(obj)
         end
         GC.start # All dead objects above will be GCed, all living strings will have age = 0
 
+        # @ivoanjo: For some weird reason, the last object sampled in the "dead_heap_samples" does not always
+        # get collected the first time, leading to a flaky spec.
+        # I was able to reproduce it with `rspec spec/datadog/profiling --seed 48141 --fail-fast` and it's
+        # kinda bizarre since e.g. if you add one more `Object.new` it also stops flaking, so is it perhaps
+        # related to Ruby's conservative GC?
+        # I've bisected this and undoing 3d4b7fcf30b529b191ca737ae13629eb27b8ab63 also makes the flakiness
+        # go away, but again, that change doesn't seem to have anything to do with GC.
+        # As the weird behavior is transitory, e.g. it provably goes away on the next GC, I'll go with this
+        # workaround for now.
```

Comment:
Yeah I seem to remember Ruby taking into account register values in its heuristic to detect usages of objects. May just be that with the extra caching of things like iseq and what not we make it slightly more likely to hit these edge cases?

Reply:
Yeah this is my suspicion -- Ruby's GC is conservative, in that if anything "looks" like a `VALUE`, it assumes it is. Actually debugging this would mean going into the GC marking code and understand why exactly it's making this decision which is somewhat of a time investment which is why I took the slightly-easier way out 😅.